graylog2-server/src/main/java/org/graylog2/outputs/MessageOutputFactory.java

```diff
@@ -44,35 +49,56 @@ public MessageOutput fromStreamOutput(Output output, final Stream stream, Config
         final String outputType = output.getType();
         Preconditions.checkArgument(outputType != null);
 
+        // We have two output lists for backwards compatibility.
+        // See comments in MessageOutput.Factory and MessageOutput.Factory2 for details
+        final MessageOutput.Factory2<? extends MessageOutput> factory2 = this.availableOutputs2.get(outputType);
```

Comment:
@bernd Minor question: Is there any reason why `Factory2` is before `Factory` here? Just wondering if we want `Factory` to come first for consistency. Same question for the if-else block that follows.

Reply:
@danotorrey The idea was that plugins using the newer factory interface should be preferred if the same plugin is registered twice for some reason. I will add a comment that explains it.